src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -596,9 +605,13 @@ def reload_module(module):
 
 def get_default_admin_username():
     try:
-        return getpass.getuser()
+        username = getpass.getuser()
     except KeyError:
-        return None
+        username = None
+    if username is None or username.lower() in DISALLOWED_USER_NAMES:
+        logger.warning('Default username %s is a reserved username. Use azureuser instead.', username)
```

Comment:
This function may be invoked multiple times. So this message may appear more than once. Not very elegant.

Reply:
I know how to fix it now.

Reply:
Fixed!